test/extensions/wasm/wasm_test.cc

```diff
@@ -77,24 +77,32 @@ TEST_P(WasmTestMatrix, Logging) {
       absl::StrCat("envoy.wasm.runtime.", std::get<0>(GetParam())), vm_id, vm_configuration, scope,
       cluster_manager, *dispatcher);
   EXPECT_NE(wasm, nullptr);
+  auto wasm_ptr = wasm.get();
+  auto wasm_handler = std::make_unique<Extensions::Common::Wasm::WasmHandle>(std::move(wasm));
```

Comment:
I know this is only a test, but we're giving away the `wasm` object here and are referring to its raw pointer in rest of the code, even after `wasm_handler.reset()`, at which points we have no guarantee that it still point to anything valid.

Reply:
In this test it has to be valid.  Let me do it with a weak_ptr.  That will assert in an appropriate way if the test fails.

Reply:
done

Reply:
I was thinking more of simply moving the `std::move` to few lines below, but that works as well. Thanks!